source/common/http/http3/conn_pool.cc

```diff
@@ -61,16 +61,12 @@ Http3ConnPoolImpl::Http3ConnPoolImpl(
     : FixedHttpConnPoolImpl(host, priority, dispatcher, options, transport_socket_options,
                             random_generator, state, client_fn, codec_fn, protocol),
       connect_callback_(connect_callback) {
-  auto source_address = host_->cluster().sourceAddress();
-  if (!source_address.get()) {
-    auto host_address = host->address();
-    source_address = Network::Utility::getLocalAddress(host_address->ip()->version());
-  }
```

Comment:
Is it possible to write a unit test which covers this? I can imagine it might not be easy in which case, the integration test covers it. But it might be nice to have a unit test. (No worries though, if this is non trivial)

Reply:
done